Sanitize repo simulator role names when writing

MVrachev opened the pull request:
**Description of the changes being introduced by the pull request**:

Sanitize the role names in the repo simulator when dumping in a
directory the same way the ngclient does.
That's necessary because when testing fishy role names from `test_updater_with_simulator` leads to an
error:
`PermissionError: [Errno 13] Permission denied: '/.json'`

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Sanitize repo simulator role names when writing

Sanitize the delegated rolenames in the repo simulator when dumping in
a directory the same way the ngclient does.
That's necessary because when testing fishy role names leads to an
error:
"PermissionError: [Errno 13] Permission denied: '/.json'"

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

jku commented:
alternatively could sanitize the delegated rolename when writing the file like client does (`quoted_role = parse.quote(role, "")`)

sechkova commented:
Well it just happened that I did the same thing yesterday: dc908eee52dcf2b66dcd38eb54c972a2b3e17ec2
(without the part in test_updater_with_simulator.py)

MVrachev commented:
> Well it just happened that I did the same thing yesterday: dc908ee
(without the part in test_updater_with_simulator.py)

@sechkova I am sorry it happened that way, I haven't read #1689 and didn't notice that. 
I will drop this commit and leave only the commit related to the error.

sechkova commented:
Bad timing on both sides 🤷 Can you have a look at my commit before dropping yours? Maybe I've missed something.

MVrachev commented:
> Bad timing on both sides 🤷 Can you have a look at my commit before dropping yours? Maybe I've missed something.

I just had a look and yours seems to cover what I did here, so don't worry.

MVrachev commented:
I updated the code by using what @jku suggested.
Additionally, the commit and pr descriptions were changed together with the pr title.

jku reviewed (changes requested):
This commit does not do what the message says.

* It url encodes role names that do not need sanitizing (but does not url encode the delegated role names that could need it)
* Then it skips dumping metadata for roles that actually would need sanitizing if they were used as filenames?

MVrachev commented:
> This commit does not do what the message says.
> 
> * It url encodes role names that do not need sanitizing (but does not url encode the delegated role names that could need it)
> * Then it skips dumping metadata for roles that actually would need sanitizing if they were used as filenames?

I changed the code to dump the delegated roles and removed the code where I skipped dumping.
